include/knowhere/factory.h

```diff
@@ -22,14 +22,14 @@ namespace knowhere {
 class IndexFactory {
  public:
     Index<IndexNode>
-    Create(const std::string& name, const Object& object = nullptr);
+    Create(const std::string& name, const std::string& version, const Object& object = nullptr);
```

Comment:
Why we need to pass the version during the build. I didn't fully the get versioning design, any description or doc about this? 
Thanks